cmd/dockerd/daemon_windows.go

```diff
@@ -93,7 +94,30 @@ func newCgroupParent(config *config.Config) string {
 	return ""
 }
 
-func (cli *DaemonCli) initContainerD(_ context.Context) (func(time.Duration) error, error) {
+func (cli *DaemonCli) initContainerD(ctx context.Context) (func(time.Duration) error, error) {
+	if cli.Config.ContainerdAddr == "" && cli.Config.DefaultRuntime == config.WindowsV2RuntimeName {
```

Comment:
Heads up that I'm changing this function to `initContainerd` (lowercase) in https://github.com/moby/moby/pull/43945. In that PR I also removed the `.Config` from these, to make it _slightly_ less verbose.

Perhaps this should use an early return here?

```go
if cli.ContainerdAddr != "" {
	// use system containerd at the given address.
	return nil, nil
}

if cli.Config.DefaultRuntime != config.WindowsV2RuntimeName {
	// containerd is only used if the V2 (io.containerd.runhcs.v1) runtime is used.
	return nil, nil
}
```

As to the `config.WindowsV2RuntimeName` check; I guess we currently use that as switch to toggle between containerd and the built-in runtime, so makes sense; slightly wondering if (in light of https://github.com/moby/moby/pull/43946), we might run into issues if alternative runtime may come to exist (not really urgent I guess).